p2p/libp2p/netMessenger.go

```diff
@@ -1276,6 +1308,19 @@ func (netMes *networkMessenger) SetPeerShardResolver(peerShardResolver p2p.PeerS
 	return nil
 }
 
+// SetCurrentBytesProvider sets the current peer bytes provider that is able to prepare the bytes to be sent to a new peer
+func (netMes *networkMessenger) SetCurrentBytesProvider(currentBytesProvider p2p.CurrentPeerBytesProvider) error {
+	if check.IfNil(currentBytesProvider) {
+		return p2p.ErrNilCurrentPeerBytesProvider
+	}
+
+	netMes.mutCurrentBytesProvider.Lock()
+	netMes.currentBytesProvider = currentBytesProvider
+	netMes.mutCurrentBytesProvider.Unlock()
```

Comment:
after setting the new currentBytesProvider let's notify all already connected peers. In this way, all other peers will be notified what is the new shard for the connected peer

Reply:
done